File: algorithm/baekjoon/codeplus/first-solve/src/main/java/com/keunwon/algorithm/datastructures/Problem1158.java
package com.keunwon.algorithm.datastructures;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class Problem1158 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();
        StringBuilder sb = new StringBuilder();
        Queue<Integer> queue = new LinkedList<>();

        for (int i = 1; i <= N; i++) {
            queue.add(i);
        }

        sb.append("<");
        while (!queue.isEmpty()) {
            for (int i = 0; i < K - 1; i++) {
                queue.offer(queue.poll());
            }

            sb.append(queue.poll());

            if (!queue.isEmpty()) {
                sb.append(", ");
            }
        }
        sb.append(">");

        System.out.println(sb.toString());
    }
}

File: algorithm/kotlin-algorithm/src/main/kotlin/com/keunwon/algorithm/baekjoon/Problem1158.kt
package com.keunwon.algorithm.baekjoon

import java.util.*

/**
 * Title: 요세푸스
 * Level: 실버-4
 **/
class Problem1158 {
    fun solution(n: Int, k: Int): String {
        val queue = LinkedList<Int>().apply { repeat(n) { offer(it + 1) } }
        val answer = mutableListOf<Int>()

        while (queue.isNotEmpty()) {
            repeat(k - 1) { queue.offer(queue.poll()) }
            answer.add(queue.poll())
        }
        return "<${answer.joinToString(", ")}>"
    }
}

fun main() {
    val (n, k) = readLine()!!.split(" ").map { it.toInt() }
    Problem1158().solution(n, k).also { println(it) }
}
